interpreter/text/parser.mly

```diff
@@ -65,6 +65,12 @@ let simd_lane_lit shape l at =
   | F32x4 -> LitPat (Values.F32 (F32.of_string l) @@ at) @@ at
   | F64x2 -> LitPat (Values.F64 (F64.of_string l) @@ at) @@ at
 
+let simd_lane_index s at =
+  try
+    let n = int_of_string s in
+    if n >= 0 && n < 256 then n else raise (Failure "")
+  with Failure _ -> error at "malformed lane index"
```

Comment:
Nit: this could be written more elegantly:
```suggestion
   match int_of_string s with
   | n when 0 <= n && n < 256 -> n
   | _ | exception Failure _ -> error at "malformed lane index"
```

Reply:
Neat! I didn't know about pattern matching on exception, thanks, please see #301 .